packages/api/src/derives/staking/stakingAccount.ts

```diff
@@ -0,0 +1,100 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import {DerivedStakingQuery} from '@polkadot/api-derive/types';
+import {memo} from '@polkadot/api-derive/util';
+import {createType, Option, TypeRegistry, Vec} from '@polkadot/types';
+import {
+  AccountId,
+  Exposure,
+  Keys,
+  Nominations,
+  RewardDestination,
+  StakingLedger,
+  ValidatorPrefs,
+} from '@polkadot/types/interfaces';
+import {ITuple} from '@polkadot/types/types';
+import {combineLatest, Observable, of} from 'rxjs';
+import {map, switchMap} from 'rxjs/operators';
+
+type MultiResultV2 = [
+  Option<AccountId>,
+  Option<Nominations>,
+  RewardDestination,
+  Exposure,
+  [ValidatorPrefs],
+  Option<Keys>,
+  Option<StakingLedger>
+];
+
+function unwrapSessionIds(
+  stashId: AccountId,
+  queuedKeys: [AccountId, Keys][],
+  nextKeys: Option<Keys>
+): {nextSessionIds: AccountId[]; sessionIds: AccountId[]} {
+  const sessionIds = (queuedKeys.find(([currentId]): boolean => currentId.eq(stashId)) || [
+    undefined,
+    [] as AccountId[],
+  ])[1];
+  const nextSessionIds = nextKeys.unwrapOr([] as AccountId[]);
```

Comment:
The unwrapping of nextSessionIds here as part of this function is rather coincidental than logical. However, as this is an internal function, I would be less fussy about that.